migtests/scripts/compare-html-reports.py

```diff
@@ -129,44 +133,85 @@ def generate_diff_list(list1, list2, section_name, file1_path, file2_path):
 def dict_to_list(dict_data):
     """Convert dictionary to list of formatted strings."""
     return [f"{k} -> {v}" for k, v in dict_data.items()]
+    
+def compare_html_tags(html_data1, html_data2):
+    """Compare the unique tags in the two HTML reports."""
 
-def compare_html_reports(file1, file2):
-    """Compares two HTML reports and prints structured differences."""
-    with open(file1, "r", encoding="utf-8") as f1, open(file2, "r", encoding="utf-8") as f2:
-        html_data1 = extract_html_data(f1.read())
-        html_data2 = extract_html_data(f2.read())
+    def get_unique_tags(html_content):
+        """Extracts all unique tag names from the given HTML content."""
+        soup = BeautifulSoup(html_content, 'html.parser')
+        return {tag.name for tag in soup.find_all()}
+    
+    tags1 = get_unique_tags(html_data1)
+    tags2 = get_unique_tags(html_data2)
+    
+    missing_tags_in_file1 = tags2 - tags1  # Tags in file2 but missing in file1
+    missing_tags_in_file2 = tags1 - tags2  # Tags in file1 but missing in file2
+
+    differences = {}
+
+    if missing_tags_in_file1:
+        differences["missing_tags_in_file1"] = "\n".join(missing_tags_in_file1)
+    if missing_tags_in_file2:
+        differences["missing_tags_in_file2"] = "\n".join(missing_tags_in_file2)
+
+    return differences
```

Comment:
One high level comment - what we(consumer of the script) interested to know is the difference between expected and actual report.

Assuming here, file1 - expected, file2- actual
Wording the output from the perspective of file2(actual) will be easier to read.

can we display in output like:
```
Missing tags in actual report:
...

Extra/new tags in actual report:
```